docs/testing.md

```diff
@@ -0,0 +1,26 @@
+## Testing APB Image
+
+### Why would we want to test the image?
+The reason that we would want to be able to integration test APB's is that they have underlying images that could have been changed and would potentially cause the APB to fail. This means that to test the APB we would like to give the author the ability to verify certain actions by calling actions with default params.
+
+### Design
+The base APB entrypoint will learn the test action. The test action will be a test harness script *(This script will run pre-defined values for now, this should be configurable in the future)*.
+
+To include the testing of an APB add to the layout and add files and example of the layout is 
+```bash
+my-apb/
+├── apb.yml
+├── Dockerfile
+├── playbooks/
+    └── provision.yml  
+└── roles/
+    └── ...
+└── test/
```

Comment:
If we're adding test as a new action, it probably makes more sense to put it into `playbooks/test.yml`. That will allow it to run without needing to change our base image behavior at all. Think it will also return a not implemented error by default if the playbook doesn't exist.